src/typecheck.rs

```diff
@@ -1630,6 +1630,32 @@ pub fn get_uop_type(state: &mut State, op: &UnaryOp) -> Result<TypeWrapper, Type
             //mk_tyw_record!(; TypeWrapper::Ptr(new_var(state.table))),
             mk_typewrapper::list(AbsType::Str())
         ),
+        // Str -> Str
+        UnaryOp::StrTrim() => mk_tyw_arrow!(mk_typewrapper::str(), mk_typewrapper::str()),
+        // Str -> List Str
+        UnaryOp::StrChars() => mk_tyw_arrow!(
+            mk_typewrapper::str(),
+            mk_typewrapper::list(mk_typewrapper::str())
+        ),
+        // Str -> Num
+        UnaryOp::CharCode() => mk_tyw_arrow!(mk_typewrapper::str(), mk_typewrapper::num()),
+        // Num -> Str
+        UnaryOp::CharFromCode() => mk_tyw_arrow!(mk_typewrapper::num(), mk_typewrapper::str()),
+        // Str -> Str
+        UnaryOp::StrUppercase() => mk_tyw_arrow!(mk_typewrapper::str(), mk_typewrapper::str()),
+        // Str -> Str
+        UnaryOp::StrLowercase() => mk_tyw_arrow!(mk_typewrapper::str(), mk_typewrapper::str()),
+        // Str -> Num
+        UnaryOp::StrLength() => mk_tyw_arrow!(mk_typewrapper::str(), mk_typewrapper::num()),
+        // Dyn -> Str
+        UnaryOp::StrFrom() => mk_tyw_arrow!(mk_typewrapper::dynamic(), mk_typewrapper::num()),
+        // Str -> Num
+        UnaryOp::NumFrom() => mk_tyw_arrow!(mk_typewrapper::str(), mk_typewrapper::num()),
```

Comment:
Fair. I choose 2, because it doesn't sound right to have a more liberal static type while we already know it will fail for anything else than a string. Maybe `NumFromStr`, or `ParseNum` as you suggest.